src/main/webapp/app/intellij/ide-build-and-test.service.ts

```diff
@@ -46,9 +46,10 @@ export class IdeBuildAndTestService {
      * Listens on any new builds for the user's participation on the websocket and forwards incoming results to the IDE
      *
      * @param exercise The exercise for which build results should get forwarded
+     * @param participation The (optional) participation to subscribe to. The default is the first student participation
      */
-    listenOnBuildOutputAndForwardChanges(exercise: ProgrammingExercise): Observable<void> {
-        const participationId = exercise.studentParticipations[0].id;
+    listenOnBuildOutputAndForwardChanges(exercise: ProgrammingExercise, participation: Participation | null = null): Observable<void> {
+        const participationId = participation ? participation.id : exercise.studentParticipations[0].id;
```

Comment:
make sure that the websocket subscription is removed when the corresponding component (that uses this service) is destroyed

Reply:
I added an additional call that removes the subscription once I have the result